src/VisualStudio/Core/Def/Implementation/TableDataSource/TableItem.cs

```diff
@@ -68,7 +74,7 @@ public TableItem(IEnumerable<TableItem<T>> items)
             }
 
             // order of item is important. make sure we maintain it.
-            _cache = SharedInfoCache.GetOrAdd(collectionHash, ordereditems, c => new SharedInfoCache(c.Select(i => i.PrimaryDocumentId).ToImmutableArray()));
+            _cache = SharedInfoCache.GetOrAdd(collectionHash, orderedItems, c => new SharedInfoCache(c.Select(i => i.PrimaryDocumentId).ToImmutableArray()));
```

Comment:
Not your doing, but it seems odd that we're caching this IEnumerable that has a bunch of queries on it. Are we re-running the OrderBy and filtering each time this cached data is used?